web/src/desktop_notifications.ts

```diff
@@ -0,0 +1,107 @@
+import $ from "jquery";
+
+import type {ElectronBridge} from "./electron_bridge";
+import type {Message} from "./message_store";
+
+const electron_bridge: ElectronBridge | undefined = window.electron_bridge;
+class ElectronBridgeNotification extends EventTarget {
+    constructor(title: string, options: NotificationOptions) {
+        super();
+        if (
+            electron_bridge &&
+            electron_bridge.new_notification &&
+            typeof electron_bridge.new_notification === "function"
+        ) {
+            Object.assign(
+                this,
+                electron_bridge.new_notification(
+                    title,
+                    options,
+                    (type: string, eventInit: EventInit) =>
+                        this.dispatchEvent(new Event(type, eventInit)),
+                ),
+            );
+        }
+    }
+    static get permission(): NotificationPermission {
+        return Notification.permission;
+    }
+    static async requestPermission(
+        callback: NotificationPermissionCallback | undefined = undefined,
+    ): Promise<NotificationPermission> {
+        if (callback) {
+            callback(await Promise.resolve(Notification.permission));
+        }
+        return Notification.permission;
+    }
+    close(): void {
+        this.close();
+    }
+}
+
+export type NoticeMemEntry = {
+    obj: ElectronBridgeNotification | Notification;
+    msg_count: number;
+    message_id: number;
+};
+
+export const notice_memory: Map<string, NoticeMemEntry> = new Map();
+
+export let NotificationAPI: typeof ElectronBridgeNotification | typeof window.Notification;
+
+if (
+    electron_bridge &&
+    electron_bridge.new_notification &&
+    typeof electron_bridge.new_notification === "function"
```

Comment:
`&& typeof electron_bridge.new_notification === "function"` is an unnecessary addition.

Reply:
Thank You for pointing it out.
I will remove that extra condition